### PR TITLE
Update only licensepools for current collection and metadata.

### PR DESCRIPTION
**Pull request by tdilauro**
## Description

This branch improves performance and eliminates unexpected collateral updates by focusing licensepool updates to only those associated with the metadata's edition  and the collection being imported.

## Motivation and Context

For each metadata record, we were updating all licensepools for the collection, which resulted in exponential work and could affect licensepools that were not associated with with a record in the metadata being processed.

~~Depends on [server_core #1242](https://github.com/NYPL-Simplified/server_core/pull/1242).~~

## How Has This Been Tested?

Created new tests. Ran new and existing tests.

## Checklist:
- N/A I have updated the documentation accordingly.
- [X] All new and existing tests passed.

### CODE REVIEW

File: scripts.py [hunk 1]
```diff
@@ -1374,11 +1374,12 @@ def run_with_arguments(
         if dry_run:
             mirrors = None
 
+        self_hosted_collection = collection_type.is_self_hosted
         replacement_policy = ReplacementPolicy.from_license_source(self._db)
         replacement_policy.mirrors = mirrors
         metadata_records = self.load_metadata(metadata_file, metadata_format, data_source_name, default_medium_type)
         for metadata in metadata_records:
-            self.work_from_metadata(
+            work, licensepool = self.work_from_metadata(
```
Comment (vbessonov) [inline scripts.py:1382]: Since `work` is not used, I think it can be replaced with `_` to not raise any unused variable warnings.

**Review by vbessonov — APPROVED** [review]
I proposed a small cosmetic change but overall everything looks good.

**Comment by tdilauro** [comment]
Based on feedback on [server_core PR #1242](https://github.com/NYPL-Simplified/server_core/pull/1242#discussion_r585593822, I will withdraw the server_core PR and rework this on.

**Comment by tdilauro** [comment]
@vbessonov Incorporated your suggestions and pulled the `core` update out.

**Review by vbessonov — APPROVED** [review]

**Comment by tdilauro** [comment]
@leonardr This is ready to merge, assuming you don't have any concerns.

**Review by leonardr — APPROVED** [review]

**Pull request merged**